pkg/provision/ec2.go

```diff
@@ -0,0 +1,190 @@
+package provision
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+)
+
+type EC2Provisioner struct {
+	ec2Provisioner *ec2.EC2
+}
+
+func NewEC2Provioner(region string) (*EC2Provisioner, error) {
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(region),
+	})
+	svc := ec2.New(sess)
+	return &EC2Provisioner{ec2Provisioner: svc}, err
+}
+
+func (p *EC2Provisioner) Provision(host BasicHost) (*ProvisionedHost, error) {
+	image, err := p.lookupAMI(host.OS)
+	if err != nil {
+		return nil, err
+	}
+
+	group, err := p.securityGroup(host.Additional["inlets-port"])
+	if err != nil {
+		return nil, err
+	}
+
+	runResult, err := p.ec2Provisioner.RunInstances(&ec2.RunInstancesInput{
+		ImageId:      image,
+		InstanceType: aws.String(host.Plan),
+		MinCount:     aws.Int64(1),
+		MaxCount:     aws.Int64(1),
+		UserData:     &host.UserData,
+		NetworkInterfaces: []*ec2.InstanceNetworkInterfaceSpecification{
+			{
+				DeviceIndex:              aws.Int64(int64(0)),
+				AssociatePublicIpAddress: aws.Bool(true),
+				DeleteOnTermination:      aws.Bool(true),
+				Groups:                   []*string{group},
+			},
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	err = p.ec2Provisioner.WaitUntilInstanceStatusOk(&ec2.DescribeInstanceStatusInput{
+		InstanceIds: []*string{runResult.Instances[0].InstanceId},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	code := int64(16)
```

Comment:
What is 16? Where is this documented on AWS? Can we link to it?

Reply:
16 is ok, I link to the docs